skrypty/Wskazniki_srodowiskowe.sql

```diff
@@ -0,0 +1,272 @@
+
+-- CZʌC FINANSOWA, WYTYPOWANIE KRAJ�W DO ANALIZY ?RODOWISKOWEJ
+-- najpierw tworzymy tabel? ograniczaj?c? dane dla lat >1989
+
+create table PKB as 
+	(select  * 
+		from bazafin  
+		where indicatorname ilike 'GDP, PPP (current international $)' and "Year" > 1989 )
+
+-- najwy?sze PKB (?rednia wszystkich lat)
+
+select shortname, avg(value) 
+  from pkb 
+group by shortname 
+order by avg(value) desc
+limit 5
+
+
+-- tutaj kraje z najwi?kszym wzrostem PKB na przestrzeni 10 lat w zadanym przedziale
+with PKB_pom as 
+	(select shortname, avg(value) as srednia, lag(avg(value),10) over (order by shortname) as count_lagged 
+	   from pkb 
+   group by shortname  
+   order by avg(value) desc
+      limit 5)

```

Comment:
Ogólnie jest dużo lepiej niż było, kod jest znacznie czytelniejszy, ale jest jeszcze parę rozjazdów względem konwencji. W zaznaczonym fragmencie kodu widać to dobrze. Po pierwsze, jeśli ubieramy coś w nawiasy, to to co jest obejmowane przez ten nawias nigdy nie powinno zaczynać się wcześniej niż ten nawias (tu np. group by i order by "wystają"). Po drugie kolumny, które wybieramy w zapytaniu rozbijamy do kolejnych linii (chyba, że wybieramy ich np. kilkadziesiąt, wtedy lepiej grupować po kilka w jednej linii, ale zawsze wyrównując do lewej strony pierwszej wybieranej kolumny). Po trzecie - gdy mamy np GROUP BY/ORDER BY, to wyrównujemy do SELECT słowo ORDER. Poniżej przykład:

```
SELECT count(kolumna_1),
       kolumna_2,
       kolumna_3
  FROM tabelka
 WHERE warunek1
   AND warunek2
 GROUP BY kolumna_2, kolumna_3
```

W paru miejscach rozjeżdzają Ci się też te wyrównania (np. SELECT i FROM są ze sobą wyrównane, ale WHERE już nie).
Tak jak pisałem Ci na slacku, to wszystko są już tylko uwagi na przyszłość i rzeczy do opcjonalnej (choć zalecanej!) poprawy. Czekamy aż Jakub wprowadzi swoje poprawki i można mergować.